aws_sagemaker_studio/sagemaker_studio_image_build/tensorflow_bring_your_own/tensorflow_bring_your_own.ipynb

```diff
@@ -0,0 +1,740 @@
+    "\n",
+    "instance_type = 'ml.m4.xlarge'\n",
+    "\n",
+    "estimator = Estimator(role=role,\n",
+    "                      train_instance_count=1,\n",
+    "                      train_instance_type=instance_type,\n",
+    "                      image_name=ecr_image,\n",
+    "                      hyperparameters=hyperparameters)\n",
+    "\n",
+    "estimator.fit(data_location)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Host our model using SageMaker Hosting"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = estimator.deploy(1, instance_type)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Making predictions using Python SDK\n",
+    "To make predictions, we use an image that is converted using OpenCV into a json format to send as an inference request. We need to install OpenCV to deserialize the image that is used to make predictions.\n",
+    "\n",
+    "The JSON reponse will be the probabilities of the image belonging to one of the 10 classes along with the most likely class the picture belongs to. The classes can be referenced from the CIFAR-10 website. Since we didn't train the model for that long, we aren't expecting very accurate results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!apt-get update\n",
+    "!apt-get install -y libsm6 libxext6 libxrender-dev libgl1-mesa-glx"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install opencv-python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cv2\n",
+    "import numpy\n",
+    "\n",
+    "from sagemaker.predictor import json_serializer, json_deserializer\n",
+    "\n",
+    "# Update the image path below based on the path to the /data/cat.png image on Studio\n",
+    "image = cv2.imread(\"data/cat.png\", 1)\n",
+    "\n",
+    "# resize, as our model is expecting images in 32x32.\n",
+    "image = cv2.resize(image, (32, 32))\n",
+    "\n",
+    "data = {'instances': numpy.asarray(image).astype(float).tolist()}\n",
+    "\n",
+    "# The request and response format is JSON for TensorFlow Serving.\n",
+    "# For more information: https://www.tensorflow.org/serving/api_rest#predict_api\n",
+    "predictor.accept = 'application/json'\n",
+    "predictor.content_type = 'application/json'\n",
+    "\n",
+    "predictor.serializer = json_serializer\n",
+    "predictor.deserializer = json_deserializer\n",
+    "\n",
+    "# For more information on the predictor class.\n",
+    "# https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/predictor.py\n",
+    "predictor.predict(data)"
```

Comment:
I got really bad results.... low scoring and it's not really showing off that I used something custom (BYO). Seems like we could point this out somehow... Identifying the custom hosting script and how you can tweak the output.